src/odemis/gui/cont/tabs.py

```diff
@@ -3153,6 +3150,15 @@ def _onSEMpxs(self, pixel_size):
         best_mpp = self._sem_view.mpp.clip(best_mpp)
         self._sem_view.mpp.value = best_mpp
 
+    def _on_stage_pos(self, pos):
+        """
+        Called when the stage is moved (and the tab is shown)
+        :param pos: (dict str->float or None) updated position of the stage
+        """
+        if not self.IsShown():
+            return
```

Comment:
It's independent of the tab being currently selected ("Shown") or not. => Remove this check.